src/components/DatePicker/index.android.js

```diff
@@ -14,50 +15,59 @@ class DatePicker extends React.Component {
         };
 
         this.showPicker = this.showPicker.bind(this);
-        this.raiseDateChange = this.raiseDateChange.bind(this);
-    }
-
-    /**
-     * @param {Event} event
-     */
-    showPicker(event) {
-        this.setState({isPickerVisible: true});
-        event.preventDefault();
+        this.setDate = this.setDate.bind(this);
     }
 
     /**
      * @param {Event} event
      * @param {Date} selectedDate
      */
-    raiseDateChange(event, selectedDate) {
+    setDate(event, selectedDate) {
         if (event.type === 'set') {
-            this.props.onChange(selectedDate);
+            this.props.onInputChange(selectedDate);
         }
 
         this.setState({isPickerVisible: false});
     }
 
+    /**
+     * @param {Event} event
+     */
+    showPicker(event) {
+        this.setState({isPickerVisible: true});
+        event.preventDefault();
+    }
+
     render() {
-        const dateAsText = this.props.value ? moment(this.props.value).format(CONST.DATE.MOMENT_FORMAT_STRING) : '';
+        const dateAsText = this.props.defaultValue ? moment(this.props.defaultValue).format(CONST.DATE.MOMENT_FORMAT_STRING) : '';
 
         return (
             <>
                 <TextInput
                     label={this.props.label}
                     value={dateAsText}
                     placeholder={this.props.placeholder}
-                    hasError={this.props.hasError}
                     errorText={this.props.errorText}
                     containerStyles={this.props.containerStyles}
                     onPress={this.showPicker}
                     editable={false}
                     disabled={this.props.disabled}
+                    onBlur={this.props.onBlur}
+                    ref={(el) => {
+                        if (!this.props.innerRef) {
+                            return;
+                        }
```

Comment:
Do we need this? If the `this.props.innerRef` fails `isFunction` below won't this all happen anyway?